depot/impl.go

```diff
@@ -0,0 +1,141 @@
+package depot
+
+import (
+	"time"
+
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"github.com/square/certstrap/depot"
+	mgo "gopkg.in/mgo.v2"
+)
+
+type DBUser struct {
+	ID   string    `bson:"_id"`
+	Cert string    `bson:"cert"`
+	TTL  time.Time `bson:"ttl"`
+}
+
+var (
+	dbUserIDKey   = bsonutil.MustHaveTag(DBUser{}, "ID")
+	dbUserCertKey = bsonutil.MustHaveTag(DBUser{}, "Cert")
+	dbUserTTLKey  = bsonutil.MustHaveTag(DBUser{}, "TTL")
+)
+
+type mongoCertDepot struct {
+	session        *mgo.Session
+	databaseName   string
+	collectionName string
+	expireAfter    time.Duration
+}
+
+type MongoCertDepotOptions struct {
+	MongoDBURI           string
+	MongoDBDialTimeout   time.Duration
+	MongoDBSocketTimeout time.Duration
+	DatabaseName         string
+	CollectionName       string
+	ExpireAfter          time.Duration
+}
+
+// Create a new cert depot in the specified MongoDB.
+func NewMongoCertDepot(opts MongoCertDepotOptions) (depot.Depot, error) {
```

Comment:
name this new mgo cert depo, and try and structure it some way that we can have a new driver implementation

Reply:
what does it mean exactly to structure for a new driver implementation?

Reply:
I think it's fine as is